libvast/src/system/disk_monitor.cpp

```diff
@@ -133,20 +133,18 @@ disk_monitor(disk_monitor_actor::stateful_pointer<disk_monitor_state> self,
           ->request(static_cast<disk_monitor_actor>(self), caf::infinite,
                     atom::erase_v)
           .then(
-            [] {
+            [=] {
+              self->state.purging = false;
```

Comment:
I think we need to solve this differently. The `erase` handler below returns after scheduling an initial set of partitions, so this `then` handler runs while the deletion is actually just starting. But analyzing the code further I realized that there is an additional problem: In case step size is greater than 1 the `erase` handler seems to turn into a fork bomb. Let's discuss tomorrow.